### PR TITLE
docs: Add recipe for TiddlyWiki on Node.js

### PR DESCRIPTION
**Pull request by favadi**
## Summary
Add the recipe documentation for using Pomerium with TiddlyWiki Node.js version.

**Checklist**:
- [ ] add related issues
- [x] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] ready for review

### CODE REVIEW

**Comment by cuonglm** [comment]
Linter error was fixed in https://github.com/pomerium/pomerium/commit/72b6347886bd4faeb12ec0025b133643de8f01ba, please rebase 😄

File: docs/recipes/tiddlywiki.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+We are using the Node.js application in this guide.
+
+## Where Pomerium fits
+
+TiddlyWiki allows a simple form of authentication by using authenticated-user-header parameter of [listen command](https://tiddlywiki.com/static/ListenCommand.html). Pomerium provides the ability to login with well-known identity provider like Google, Github...
+
+## Pre-requisites
+
+This guide assumes you have already completed one of the [quick start] guides, and have a working instance of Pomerium up and running. For purpose of this guide, I'm going to use docker-compose, though any other deployment method would work equally well.
```
Comment (cuonglm) [inline docs/recipes/tiddlywiki.md:30]:
```suggestion
This guide assumes you have already completed one of the [quick start] guides, and have a working instance of Pomerium up and running. For purpose of this guide, we're going to use docker-compose, though any other deployment method would work equally well.
```

File: docs/recipes/tiddlywiki.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+### Pomerium Config
+
+```yaml
+- from: https://wiki.example.local
+  to: http://tiddlywiki:8080
+  allowed_users:
+    - reader1@example.com    
+    - writer1@example.com    
+```
+### Docker-compose
```
Comment (desimone) [inline docs/recipes/tiddlywiki.md:43]: This is great. Do you mind adding this example to https://github.com/pomerium/pomerium/tree/master/examples ?

File: docs/recipes/tiddlywiki.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+We are using the Node.js application in this guide.
+
+## Where Pomerium fits
+
+TiddlyWiki allows a simple form of authentication by using authenticated-user-header parameter of [listen command](https://tiddlywiki.com/static/ListenCommand.html). Pomerium provides the ability to login with well-known identity provider like Google, Github...
+
+## Pre-requisites
+
+This guide assumes you have already completed one of the [quick start] guides, and have a working instance of Pomerium up and running. For purpose of this guide, We're going to use docker-compose, though any other deployment method would work equally well.
```
[the remark this answers is not in the document]
Comment (favadi) [reply docs/recipes/tiddlywiki.md]: fixed in latest commit.

File: examples/tiddlywiki/config.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+authenticate_service_url: https://authenticate.localhost.pomerium.io
+autocert: true
+autocert_use_staging: true
+idp_provider: google
+idp_client_id: REPLACEME
+idp_client_secret: REPLACEME
+
```
[the remark this answers is not in the document]
Comment (desimone) [reply examples/tiddlywiki/config.yaml:7]:
```
pomerium_1         | {"level":"debug","service":"envoy","name":"router","time":"2020-07-30T02:00:30Z","message":"\"[C27][S8211966144826854990] router decoding headers:\\n\\':method\\', \\'GET\\'\\n\\':path\\', \\'/\\'\\n\\':scheme\\', \\'http\\'\\n\\':authority\\', \\'wiki.imac.bdd.io\\'\\n\\'accept-encoding\\', \\'deflate, gzip\\'\\n\\'authority\\', \\'wiki.imac.bdd.io\\'\\n\\'pragma\\', \\'no-cache\\'\\n\\'cache-control\\', \\'no-cache\\'\\n\\'upgrade-insecure-requests\\', \\'1\\'\\n\\'user-agent\\', \\'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.89 Safari/537.36\\'\\n\\'accept\\', \\'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9\\'\\n\\'sec-fetch-site\\', \\'cross-site\\'\\n\\'sec-fetch-mode\\', \\'navigate\\'\\n\\'sec-fetch-user\\', \\'?1\\'\\n\\'sec-fetch-dest\\', \\'document\\'\\n\\'accept-language\\', \\'en-US,en;q=0.9\\'\\n\\'dnt\\', \\'1\\'\\n\\'cookie\\', \\'\\'\\n\\'x-forwarded-proto\\', \\'https\\'\\n\\'x-request-id\\', \\'ca3f0235-7bef-4c0c-ae90-659664a8bb07\\'\\n\\'x-pomerium-claim-email\\', \\'bdd@pomerium.com\\'\\n\\'x-pomerium-jwt-assertion\\', \\'eyJhbGciOiAiRVMyNTYifQ.-rngr918fuv_z_xxQhk_bL4wA\\'\\n\\'x-envoy-expected-rq-timeout-ms\\', \\'30000\\'\""}
```

Pomerium is letting the request through, 401 looks to be coming from the wiki.

File: docs/recipes/tiddlywiki.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+- from: https://wiki.example.local
+  to: http://tiddlywiki:8080
+  allowed_users:
+    - reader1@example.com    
+    - writer1@example.com    
+```
+### Docker-compose
+
+```yaml
+# docker-compose.yaml
```
[the remark this answers is not in the document]
Comment (favadi) [reply docs/recipes/tiddlywiki.md:46]: did that in latest commit.

File: examples/tiddlywiki/config.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (favadi) [reply examples/tiddlywiki/config.yaml:7]: I forgot to re copy/paste the docker-compose in the docs, the reason is tiddlywiki downcase all headers, therefore the correct setting is `- authenticated-user-header=x-pomerium-claim-email`. Fixed in latest commit.

[the remark this answers is not in the document]
Comment (desimone) [reply examples/tiddlywiki/config.yaml:7]: Cool, that did the trick. In fairness, they should not be using case sensitive headers anyway which is not part of the spec!

From RFC 2616 - "Hypertext Transfer Protocol -- HTTP/1.1", Section 4.2, "Message Headers":

> Each header field consists of a name followed by a colon (":") and the field value. Field names are case-insensitive.

[the remark this answers is not in the document]
Comment (favadi) [reply examples/tiddlywiki/config.yaml:7]: > In fairness, they should not be using case sensitive headers anyway which is not part of the spec!

Sure, created a PR for this: https://github.com/Jermolene/TiddlyWiki5/pull/4780

**Pull request merged**